### PR TITLE
Inherit exposures from parent class

### LINKED ISSUE #57: Exposures are not inherited in sub-classes
I had to do the following workaround to make this work:

``` ruby
def self.inherited(klass)
  super
  exposures.exposures.each do |name, exposure|
    klass.exposures.exposures[name] = exposure.dup
  end
end
```

### PR DESCRIPTION
**Pull request by GustavoCaso**
Fixes #57 

@solnic hope this help :)

### CODE REVIEW

**Comment by GustavoCaso** [comment]
@timriley if you want to have a look, hope the implementation works for everyone :)

File: lib/dry/view/exposures.rb [hunk 1]
```diff
@@ -40,6 +40,14 @@ def locals(input)
         }
       end
 
+      def each(&block)
+        exposures.each(&block)
+      end
+
+      def import(name, exposure)
```
Comment (timriley) [inline lib/dry/view/exposures.rb]: Would you mind moving this method underneath `#add`? Feels like the right ordering to me.

File: lib/dry/view/exposures.rb [hunk 2]
```diff
@@ -40,6 +40,14 @@ def locals(input)
         }
       end
 
+      def each(&block)
```
Comment (timriley) [inline lib/dry/view/exposures.rb]: Would you mind moving this one under `#[]`?

**Comment by waiting-for-dev** [comment]
Surely a crazy idea but, have do you think about inheriting exposure options?

```ruby
class ParentController < Dry::View::Controller
  expose(:title, layout: true, decorate: false)
  # ...

class Child1Controller < ParentController
  expose(:title) { "Title 1" }
  # ...

class Child2Controller < ParentController
  expose(:title) { "Title 2" }
  # ...
```
```erb
# application.html.erb
<title><%= title %></title>
```

**Comment by timriley** [comment]
Hi @waiting-for-dev, that's clever idea but I'm not a super fan of it, unfortunately — I think it's too implicit, and could surprise developers. The current behaviour — when an exposure is defined, it must also define all its necessary options — feels much clearer to me.

However, I still think there's a solution for your use case: in your superclass, define the exposure _without a block_, so that it refers to an instance method. Then your subclasses can provide their own matching methods, and you get the same exposure behaviour in all of them.

e.g.

```ruby
require "slim"

class BaseView < Dry::View::Controller
  config.paths = Pathname(__dir__).realpath.join("spec/fixtures/templates")
  config.layout = "test_title"

  expose :title, layout: true
end

class View < BaseView
  config.template = "hello"

  private

  def title
    "My view"
  end
end

View.new.().to_s
# => "<html><head><title>My view</title></head><body><yield></yield></body></html>"
```

**Comment by waiting-for-dev** [comment]
Cool, your solution is much clearer. Thanks @timriley :)

**Pull request merged**